synapse/storage/databases/main/relations.py

```diff
@@ -946,13 +946,19 @@ async def get_thread_id(self, event_id: str) -> str:
         Get the thread ID for an event. This considers multi-level relations,
         e.g. an annotation to an event which is part of a thread.
 
+        It only searches up the relations tree, i.e. it only considers events
+        which are the parent of the given event.
```

Comment:
What does "parent" mean here? Is P a parent of E if P belongs to E's list of `prev_events`? Or do you mean that E marks itself as relating to P?

Reply:
It is in reference to relations. Note that [the spec](https://spec.matrix.org/v1.4/client-server-api/#forming-relationships-between-events) uses the parent/child terminology. Do you have suggestions for clearer terms?

Reply:
Thoughts:
- parent/child
- relatee/relator (urgh)
- stimulus/reaction
- relation target/relation source

None of these are great though. Possibly "relation-parent"? (I've used the phrasing "auth-parent" elsewhere in reference to `auth_events`)

Another idea might be to rephrase this to avoid references ancestry, e.g.

```suggestion
        It only searches up the relations tree, i.e. it only looks for events
        that the given event cites in its `m.relates_to` content field.
```

or 

```suggestion
        It only searches up the relations tree, i.e. it only considers events
        which the given event declares itself as being related to.
```

Reply:
Neither of those are really correct though, because it also looks at that recursively? I"ll try to play with it.

Reply:
> Neither of those are really correct though, because it also looks at that recursively? I"ll try to play with it.

True, but I think the current wording also suffers from this.